org.OpenT2T.Onboarding.VeraHub/node/hub.js

```diff
@@ -0,0 +1,307 @@
+            getOptions.agent = keepAliveAgent;
+        }
+
+        var request = https.get(getOptions, function(response)  {
+            var body = '';
+            response.setEncoding('utf8');
+            response.on('data', function(data) {
+                body += data;
+            });
+
+            response.on('end', function() {
+                if (response.statusCode != 200) {
+                    deferred.reject(response);
+                } else {
+                    if (returnRawBody)
+                    {
+                        deferred.resolve(body);
+                    }
+                    else
+                    {
+                        var parsedBody = JSON.parse(body);
+                        deferred.resolve(parsedBody);
+                    }
+                }
+            });
+
+            response.on('error', function(e) {
+                deferred.reject(e);
+            });
+        });
+
+        request.on('error', (e) => {
+            deferred.reject(e);
+        });
+
+        return deferred.promise;
+    },
+};
+
+module.exports = 
+{
+    inputNeeded : [
+        {
+            type: 'input',
+            name: 'pkDevice',
+            message: 'Vera Device Serial Number (8 digit numeric): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid Serial Number.';
+                }
+            }
+        },
+        {
+            type: 'input',
+            name: 'username',
+            message: 'Vera User Name (create this in the Vera app): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid User Name.';
+                }
+            }
+        },
+        {
+            type: 'password',
+            name: 'password',
+            message: 'Vera Password (create this in the Vera app): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid Password.';
+                }
+            }
+        }
+    ],
+
+    // complete info as returned by the server
+    hubInfo: undefined,
+
+    // devices list as returned by the server 
+    devices: undefined,
+
+    // this is the server url to use when talking to the hub 
+    relayServer: undefined,
+
+    // the session token for the relay server
+    relaySessionToken: undefined,
+
+    // the is the serial number of the vera hub device
+    // it is referred to as pkDevice in the vera forums
+    pkDevice: undefined,
+
+    // authorizes to the server based on the properties saved in the object 
+    authorize: function(errorCallback) {
```

Comment:
This function (and the next one) returns a promise and also has an error callback. There should be no need for the callback: if the caller wants to handle errors then they can do that via the promise rejection.